module4/ex17_formatted_cart/ex17_formatted_cart.py

```diff
@@ -1,9 +1,30 @@
 class Cart:
-    pass
+    def __init__(self):
+        self.items = []
+
+    def add(self, item):
+        self.items.append(item)
+
+    def __format__(self, format_spec):
+        if format_spec == 'short':
+            new_list = sorted([item.name for item in self.items])
+            return ', '.join(new_list)
+        elif format_spec == 'long':
+            return '\n'.join(str(item) for item in sorted(self.items, key=lambda x: x.name))
+        else:
+            raise TypeError("unsupported format string passed to Cart.__format__")
 
 
 class Item:
-    pass
+    def __init__(self, quantity, measure, name, price):
+        self.quantity = quantity
+        self.measure = measure
+        self.name = name
+        self.price = price
+
+    def __str__(self):
+        return f"{self.quantity:5} {self.measure:5} {self.name:10} @ ${float(self.price)}...$" \
```

Comment:
This is a good opportunity to make use of `try/except` and log an error message (if `price` is not a number or a string representing a number). You can then quit the program by throwing a `ValueError` exception.